src/app/pages/PerpetualPage/components/TradingChart/index.tsx

```diff
@@ -0,0 +1,165 @@
+        library_path: '/charting_library/', //relative path of library in /public folder
+        locale: 'en',
+        load_last_chart: true, //last chart layout (if present)
+        enabled_features: [
+          'study_templates', //remove to disable storing of indicators
+          'side_toolbar_in_fullscreen_mode',
+        ], //full list of features here: https://github.com/tradingview/charting_library/wiki/Featuresets/c2ec34605fa84781048d32b87a2b08ef1466639a
+        disabled_features: [
+          'header_symbol_search',
+          //'header_saveload', //uncomment to disable storing of drawings
+          'header_compare',
+          'volume_force_overlay',
+        ],
+        overrides: {
+          'paneProperties.background': '#000000',
+          'paneProperties.vertGridProperties.color': '#454545',
+          'paneProperties.horzGridProperties.color': '#454545',
+          'symbolWatermarkProperties.transparency': 90,
+          'scalesProperties.textColor': '#AAA',
+          'scalesProperties.backgroundColor': '#121214',
+          'paneProperties.topMargin': 15,
+          'paneProperties.bottomMargin': 10,
+        },
+        autosize: true,
+        toolbar_bg: '#000000',
+        header_bg: '#000000',
+        theme,
+        time_frames: [
+          { text: '1d', resolution: '15', description: '1d', title: '1d' },
+          { text: '3d', resolution: '30', description: '3d', title: '3d' },
+          { text: '7d', resolution: '60', description: '7d', title: '7d' },
+          { text: '3m', resolution: '120', description: '3m', title: '3m' },
+          // { text: '6m', resolution: '240', description: '6m', title: '6m' },
+          // { text: '1y', resolution: '1W', description: '1y', title: '1y' },
+          { text: '5y', resolution: '1W', description: '5y', title: '5y' },
+        ],
+      };
+
+      const myChart = new widget(widgetOptions);
+      setChart(myChart);
+      myChart.onChartReady(() => {
+        setHasCharts(true);
+      });
+
+      return () => {
+        myChart.remove();
+        setHasCharts(false);
+        setChart(null);
+      };
+    } catch (e) {
+      console.error(e);
+      setHasCharts(false);
+    }
+
+    // run only once after mount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useLayoutEffect(() => {
+    if (chart && hasCharts) {
+      chart.chart().setSymbol(symbol, noop);
+    }
+  }, [chart, hasCharts, symbol]);
+
+  return (
+    <div
+      className={classNames(
+        'tw-w-full tw-h-full tw-flex tw-rounded tw-overflow-hidden',
+        hasCharts && 'tw-border',
+      )}
+      style={
+        hasCustomDimensions ? undefined : { minWidth: 270, minHeight: 500 }
+      }
+    >
+      <>
+        <div
+          id="tv_chart_container"
+          className={classNames(
+            'tv-chart-container tw-flex-grow',
+            !hasCharts && 'tw-hidden',
+          )}
+        />
+        <div
```

Comment:
I would extract this whole `div` to a separate component (e.g. `TradingChartSkeleton`) that will accept 1 property (`hasCharts`).